commons/src/main/java/io/hyscale/commons/framework/events/model/HyscaleGenericEvent.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.framework.events.model;
+
+import org.springframework.core.ResolvableType;
+import org.springframework.core.ResolvableTypeProvider;
+
+/**
+ * Class to define Generic events
+ * All events which uses Generic type should extend this class
+ *
+ */
+public abstract class HyscaleGenericEvent extends HyscaleEvent implements ResolvableTypeProvider {
```

Comment:
With the finalized list, where does this fit into ?

Reply:
Based on the discussion yesterday we won't need this at the moment but kept it in case it could be used later.